src/droplet.F

```diff
@@ -527,23 +617,49 @@ subroutine droplet_driver(dt,xh,uh,ruh,xf,yh,vh,rvh,yf,zh,mh,rmh,zf,mf,zs,    &
 !
 !----------------------------------------------------------------------
 
-      end subroutine droplet_driver
+  t_s = mpi_wtime()
 
-      subroutine BE_integration(dt,np,pdata,pdata_locind,xf,yf,x3d,y3d, &
+!----------------------------------------------------------------------
+! Final thing: Inject droplets which will be active on the next call of droplet_driver
+! Requires inputting the current 10-meter wind speed
+     
+     s10_xym = 0.0
+     do j=1,nj
+     do i=1,ni
+        s10_xym = s10_xym + s10(i,j)
+     end do
+     end do
+     call mpi_allreduce(mpi_in_place,s10_xym,1,mpi_real,mpi_sum,mpi_comm_world,ierr)
```

Comment:
Again, why do we need an `mpi_allreduce` call here?

Reply:
All of the processors need to know the value of `s10_xym`. This is the 10-meter wind speed that goes into the droplet injection routine and determines how many droplets need to be produced.

Reply:
But this means all the processors will know the maximum 10-m wind speed over the whole domain. Should it be enough for each processor to know the 10-m wind speed of its own sub-domain?

Reply:
Strictly speaking, it should really be the domain-wide average. In practice, it probably wouldn't make a big difference if we did it locally, but this would introduce a dependence on `numprocs` since different processor numbers would lead to different areas that this averaging is being done over.

Reply:
thanks for the clarification.